Expose operator lighter in SVGFECompositeElement IDL

nullhook opened the pull request:
#### 89a77dce2e2a81bf5aa6f6b6b670373ed27ea31b
<pre>
Expose operator lighter in SVGFECompositeElement IDL
<a href="https://bugs.webkit.org/show_bug.cgi?id=304105">https://bugs.webkit.org/show_bug.cgi?id=304105</a>
<a href="https://rdar.apple.com/problem/166704079">rdar://problem/166704079</a>

Reviewed by Said Abou-Hallawa.

As defined in the Compositing and Blending spec [1], the &apos;lighter&apos;
operator is specified.

This PR, as discussed in [2], exposes the &apos;lighter&apos; value to IDL.

[1] <a href="https://www.w3.org/TR/compositing-1/#porterduffcompositingoperators_plus">https://www.w3.org/TR/compositing-1/#porterduffcompositingoperators_plus</a>
[2] w3c/svgwg#424

* LayoutTests/imported/w3c/web-platform-tests/svg/types/scripted/SVGAnimatedEnumeration-SVGFECompositeElement-expected.txt:

* LayoutTests/svg/dom/SVGAnimatedEnumeration-SVGFECompositeElement-expected.txt: Removed.
* LayoutTests/svg/dom/SVGAnimatedEnumeration-SVGFECompositeElement.html: Removed.

Removed this test because it is duplicated in WPT.

* Source/WebCore/svg/SVGFECompositeElement.h:
(WebCore::SVGIDLEnumLimits&lt;CompositeOperationType&gt;::highestExposedEnumValue): Deleted.
* Source/WebCore/svg/SVGFECompositeElement.idl:
* Source/WebCore/svg/properties/SVGDecoratedEnumeration.h:
* Source/WebCore/svg/properties/SVGPropertyTraits.h:
(WebCore::SVGIDLEnumLimits::highestExposedEnumValue): Deleted.

Canonical link: <a href="https://commits.webkit.org/304941@main">https://commits.webkit.org/304941@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2d4b38a26f74b64eda40b3e088b555565aeec79b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/136916 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/9276 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/48203 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/144655 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/89889 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/a982a1a0-6dac-4edc-abb6-2f1cb2e6d338) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/138788 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/9979 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/9122 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/104694 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/76402 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/643d70b9-aae8-4acc-abad-7c79e00f6934) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/139861 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/7306 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/122675 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/85532 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/68dc6021-5981-48a9-8f94-d8e680fbabc1) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/6943 "Passed tests") | [❌ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/4643 "Found 2 new API test failures: TestWebKitAPI.WKWebExtensionAPIStorage.SetAccessLevelTrustedAndUntrustedContexts, TestWebKitAPI.PermissionsAPI.GeolocationPermissionDeniedFromPromptAndGeolocationRequestedSinceLoad (failure)") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/5242 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/116276 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/40854 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/147410 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/8959 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/41424 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/113052 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/8977 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/7528 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/113382 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28809 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/6866 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/118954 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/63176 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/9007 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/37008 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/8728 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/72573 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/8948 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/8799 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

shallawa commented on Source/WebCore/svg/SVGFECompositeElement.h:
```diff
@@ -30,7 +30,7 @@
 
 template<>
-inline unsigned SVGIDLEnumLimits<CompositeOperationType>::highestExposedEnumValue() { return enumToUnderlyingType(CompositeOperationType::FECOMPOSITE_OPERATOR_ARITHMETIC); }
+inline unsigned SVGIDLEnumLimits<CompositeOperationType>::highestExposedEnumValue() { return enumToUnderlyingType(CompositeOperationType::FECOMPOSITE_OPERATOR_LIGHTER); }
```
`SVGIDLEnumLimits` can be removed now. It was added only for `CompositeOperationType`. All calls to `SVGIDLEnumLimits<EnumType>::highestExposedEnumValue()` should be replaced with `SVGPropertyTraits<EnumType>::highestEnumValue()`.

shallawa commented on Source/WebCore/svg/SVGFECompositeElement.idl:
```diff
@@ -35,6 +35,7 @@
     const unsigned short SVG_FECOMPOSITE_OPERATOR_ATOP       = 4;
     const unsigned short SVG_FECOMPOSITE_OPERATOR_XOR        = 5;
     const unsigned short SVG_FECOMPOSITE_OPERATOR_ARITHMETIC = 6;
+    const unsigned short SVG_FECOMPOSITE_OPERATOR_LIGHTER = 7;
```
I am not sure what is more preferable here regarding indentation. But I think most of the SVG IDL use indentation for enum values. So let's have the new value also indented.

nullhook committed:
Expose operator lighter in SVGFECompositeElement IDL
https://bugs.webkit.org/show_bug.cgi?id=304105
rdar://problem/166704079

Reviewed by Said Abou-Hallawa.

As defined in the Compositing and Blending spec [1], the 'lighter'
operator is specified.

This PR, as discussed in [2], exposes the 'lighter' value to IDL.

[1] https://www.w3.org/TR/compositing-1/#porterduffcompositingoperators_plus
[2] w3c/svgwg#424

* LayoutTests/imported/w3c/web-platform-tests/svg/types/scripted/SVGAnimatedEnumeration-SVGFECompositeElement-expected.txt:

* LayoutTests/svg/dom/SVGAnimatedEnumeration-SVGFECompositeElement-expected.txt: Removed.
* LayoutTests/svg/dom/SVGAnimatedEnumeration-SVGFECompositeElement.html: Removed.

Removed this test because it is duplicated in WPT.

* Source/WebCore/svg/SVGFECompositeElement.h:
(WebCore::SVGIDLEnumLimits<CompositeOperationType>::highestExposedEnumValue): Deleted.
* Source/WebCore/svg/SVGFECompositeElement.idl:
* Source/WebCore/svg/properties/SVGDecoratedEnumeration.h:
* Source/WebCore/svg/properties/SVGPropertyTraits.h:
(WebCore::SVGIDLEnumLimits::highestExposedEnumValue): Deleted.

Canonical link: https://commits.webkit.org/304941@main
